x/authz/simulation/operations.go

```diff
@@ -108,9 +107,10 @@ func SimulateMsgGrant(ak authz.AccountKeeper, bk authz.BankKeeper, _ keeper.Keep
 			return simtypes.NoOpMsg(authz.ModuleName, TypeMsgGrant, "spend limit is nil"), nil, nil
 		}
 
-		expiration := simtypes.RandTimestamp(r)
-		if expiration.Before(ctx.BlockTime()) {
-			return simtypes.NoOpMsg(authz.ModuleName, TypeMsgGrant, "past time"), nil, nil
+		var expiration *time.Time
+		t1 := simtypes.RandTimestamp(r)
+		if !t1.Before(ctx.BlockTime()) {
```

Comment:
bummer! calling a rand here will cause sim tests to fail :( This is because our simulation tests don't simulate much - they are almost 'pre defined': we have one grant message, one exec and one revoke. it's basically one premade scenario with lot of boilerplate.